incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -91,22 +95,53 @@ func (r *ObjectReconciler) SyncNamespace(ctx context.Context, log logr.Logger, n
 	return nil
 }
 
+// SyncCluster can be called manually by the ConfigReconciler when the mode of a type is changed.
+// It enqueues all the current objects in all namespaces.
+func (r *ObjectReconciler) SyncCluster(ctx context.Context, log logr.Logger) error {
+	log = log.WithValues("gvk", r.GVK)
+
+	keys := r.Forest.GetNamespaceNames()
+	for _, ns := range keys {
+		// Enqueue all the current objects in the namespace.
+		if err := r.enqueueLocalObjects(ctx, log, ns); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 // GetGVK provides GVK that is handled by this reconciler.
 func (r *ObjectReconciler) GetGVK() schema.GroupVersionKind {
 	return r.GVK
 }
 
+// GetMode provides mode of objects that are handled by the reconciler.
+func (r *ObjectReconciler) GetMode() api.SynchronizationMode {
+	return r.Mode
+}
+
+// SetMode provides mode of objects that are handled by the reconciler.
+func (r *ObjectReconciler) SetMode(mode api.SynchronizationMode) {
+	r.Mode = mode
+}
+
 func (r *ObjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if ex[req.Namespace] {
 		return ctrl.Result{}, nil
 	}
 
+	log := r.Log.WithValues("trigger", req.NamespacedName)
+	if r.Mode == api.Ignore {
+		log.Info("Ignore reconciliation.", "gvk", r.GVK)
```

Comment:
This should be log.V(1).Info at best, but you can probably just remove it too.

Reply:
Makes sense. Removed the log.